lib/bitrange.h

```diff
@@ -65,13 +65,31 @@ class JSONLoader;
 namespace BitRange {
 
 namespace Detail {
+#if defined(__GNUC__) || defined(__clang__)
+/**
+ * @return boolean indicating if @p value is a power of two.
+ *
+ * This function is declared constexpr to allow it to be used in tests to optimize code generation.
+ */
+constexpr bool isPowerOfTwo(int value) { return value && !(value & (value - 1)); }
```

Comment:
What is the intended behavior wrt negative numbers? This currently returns `false` for e.g. `-2`, but it takes `int` as an argument. If negatives are not expected, I'd go with explicit `unsigned` here, or maybe even something like this?

```c++
template <typename T, typename = std::enable_if_t<std::is_unsigned_v<T>>>
[[nodiscard]] constexpr inline bool isPowerOfTwo(T value) noexcept {
 return (value != 0) && ((value & (value - 1)) == 0);
}
```

Reply:
For divideFloor/moduloFloor, I think it makes sense to restrict the divisor to unsigned numbers. The dividend _does_ need to be signed, but I don't call isPowerOfTwo or log2ForPowerOfTwo on the dividend.